xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp

```diff
@@ -162,6 +162,23 @@ bool CRendererBase::Configure(const VideoPicture& picture, float fps, unsigned o
   m_fps = fps;
   m_renderOrientation = orientation;
 
+  if (picture.hasDisplayMetadata || picture.color_primaries == AVCOL_PRI_BT2020)
+  {
+    bool hdr_capable, hdr_enabled;
+
+    DX::DeviceResources::Get()->DetectDisplayHDRcapable(hdr_capable, hdr_enabled);
+
+    if (hdr_enabled)
+    {
+      DX::DeviceResources::Get()->Set_HDR_MetaData(picture.displayMetadata, picture.lightMetadata);
```

Comment:
HDR metadata can be changed during playback or absent on start. Also different scenes can use different meta. why you set meta only on configuring? It must be setted for each frame in `CRendererBase::Render`.

Reply:
I disagree. It seems that due to some Microsoft bug or feature if the static HDR metadata is not configured at the beginning then they are ignored dynamic metadata too.

It is MUST to configure static metadata and optional dynamic metadata.